src/include/ipc/dai-imx.h

```diff
@@ -0,0 +1,33 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ *
+ * Copyright 2019 NXP
+ *
+ * Author: Daniel Baluta <daniel.baluta@nxp.com>
+ */
+
+#ifndef __IPC_DAI_IMX_H__
+#define __IPC_DAI_IMX_H__
+
+#include <ipc/header.h>
+#include <stdint.h>
+
+/* ESAI Configuration Request - SOF_IPC_DAI_ESAI_CONFIG */
+struct sof_ipc_dai_esai_params {
+	struct sof_ipc_hdr hdr;
+
+	/* MCLK */
+	uint16_t reserved1;
+	uint16_t mclk_id;
+	uint32_t mclk_rate; /* MCLK frequency in Hz */
+	uint32_t mclk_direction;
+
+	/* TDM */
+	uint32_t tdm_slots;
+	uint32_t rx_slots;
+	uint32_t tx_slots;
+	uint16_t tdm_slot_width;
```

Comment:
if yes to the question above you'd need to have separate fields for tdm_rx_slot_num and tdm_tx_slot_num, tdm_rx_slot_width, tdm_tx_slot_width.

Reply:
Let me check again. I'm using the same parameters as ASoC driver. Also, this follows the convention from:

``
nt (*set_tdm_slot)(struct snd_soc_dai *dai,
»       »       unsigned int tx_mask, unsigned int rx_mask,
»       »       int slots, int slot_width);
``